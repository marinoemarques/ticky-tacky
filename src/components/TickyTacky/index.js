import * as React from 'react'

import styled from 'styled-components'

import Board from '../Board'

export const Header = styled.h1`
  font-size: 2.4rem;
  line-height: 3;
`
Header.displayName = 'Header'

export const Ticky = styled.span`
  color: Crimson;

  &:before {
    content: 'Ticky'
  }
`
Ticky.displayName = 'Ticky'

export const Tacky = styled.span`
  color: SlateBlue;

  &:before {
    content: 'Tacky'
  }
`
Tacky.displayName = 'Tacky'

export default function TickyTacky () {
  return (
  <>
    <Header>
      <Ticky /><Tacky />
    </Header>
    <Board />
  </>
  )
}
