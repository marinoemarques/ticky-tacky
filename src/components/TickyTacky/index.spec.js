import * as React from 'react'

import { shallow } from 'enzyme'

import TickyTacky, { Header, Tacky, Ticky } from './'

describe('Header', () => {
  it('matches the snapshot', () => {
    expect(toJson(shallow(<Header />).first().render())).toMatchSnapshot()
  })
})

describe('Tacky', () => {
  it('matches the snapshot', () => {
    expect(toJson(shallow(<Tacky />).first().render())).toMatchSnapshot()
  })
})

describe('Ticky', () => {
  it('matches the snapshot', () => {
    expect(toJson(shallow(<Ticky />).first().render())).toMatchSnapshot()
  })
})

describe('TickyTacky', () => {
  it('matches the snapshot', () => {
    expect(toJson(shallow(<TickyTacky />))).toMatchSnapshot()
  })
})
