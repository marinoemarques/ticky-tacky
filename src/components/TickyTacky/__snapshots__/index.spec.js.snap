// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
.c0 {
  font-size: 2.4rem;
  line-height: 3;
}

<h1
  class="c0"
/>
`;

exports[`Tacky matches the snapshot 1`] = `
.c0 {
  color: SlateBlue;
}

.c0:before {
  content: 'Tacky';
}

<span
  class="c0"
/>
`;

exports[`Ticky matches the snapshot 1`] = `
.c0 {
  color: Crimson;
}

.c0:before {
  content: 'Ticky';
}

<span
  class="c0"
/>
`;

exports[`TickyTacky matches the snapshot 1`] = `
<Fragment>
  <Header>
    <Ticky />
    <Tacky />
  </Header>
  <Board />
</Fragment>
`;
